### PR TITLE
Add lint, format, package action

### PR DESCRIPTION
**Pull request by csmith14**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v2.1.2
+        with:
+          node-version: "17"
+
+      - name: Install dependencies
+        run: npm i
+
+      - name: Install rc-apps cli
+        run: npm i -g @rocket.chat/apps-cli
```
Comment (Shailesh351) [inline .github/workflows/main.yml]: @csmith14 We should use our fork of `apps-cli` here https://github.com/WideChat/Rocket.Chat.Apps-cli

Comment (csmith14) [reply .github/workflows/main.yml]: The [WideChat/Rocket.Chat.Apps-cli](https://github.com/WideChat/Rocket.Chat.Apps-cli) package fails to install, with a `TAR_ENTRY_ERROR ENOENT`:

https://github.com/WideChat/Apps.Salesforce.LiveAgent/runs/6512327350?check_suite_focus=true

image

**Comment by csmith14** [comment]
@ear-dev @Shailesh351 
After adding a step to create and activate virtual env, install of WideChat/Rocket.Chat.Apps-cli still fails:
https://github.com/WideChat/Apps.Salesforce.LiveAgent/runs/6576663171?check_suite_focus=true

**Comment by Shailesh351** [comment]
@csmith14 In that case if it's a success with the RC version of apps-cli then we can go ahead with that.

I'm not sure how it's successful. Mostly in the DF repo, it won't work because we are using some methods that are only in our version. If it's not working in DF then we will need to fix this issue.

**Comment by ear-dev** [comment]
@csmith14 I'm a little confused as to why it works from the Dialogflow repo but not in this repo..... 
 https://github.com/WideChat/Apps.Dialogflow/blob/master/.github/workflows/main.yml#L18

Could it be the difference in the node versions?  @csmith14 can you try to setup node and install the apps-cli the same way it's being done in the DF repo?

**Comment by csmith14** [comment]
Looks like the issue was the version of apps-engine in devDependencies?

**Comment by ear-dev** [comment]
> Looks like the issue was the version of apps-engine in devDependencies?

Oh interesting......  makes sense.... Probably we should update that.  Currently we're on `v1.31.0.widechat-1`

**Comment by Shailesh351** [comment]
@csmith14 @ear-dev Action is a success now but the time is around 34 minutes, earlier It was 1 minute only. Not sure how. Did we change anything else?

**Comment by ear-dev** [comment]
No idea why these install steps are taking so long.....  but I'm ok with merging this for now.

**Review by Shailesh351 — APPROVED** [review]

**Pull request merged**